app/controllers/app.py

```diff
@@ -1065,3 +1091,70 @@ def delete(self, app_id):
 
         except Exception as e:
             return dict(status='fail', message=str(e)), 500
+
+
+class AppCpuUsageView(Resource):
+    @jwt_required
+    def post(self, project_id, app_id):
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        app_memory_schema = MetricsSchema()
+        app_cpu_data = request.get_json()
+
+        validated_query_data, errors = app_memory_schema.load(app_cpu_data)
+
+        if errors:
+            return dict(status='fail', message=errors), 400
+
+        project = Project.get_by_id(project_id)
+
+        if not project:
+            return dict(
+                status='fail',
+                message=f'project {project_id} not found'
+            ), 404
+
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
+            return dict(status='fail', message='unauthorised'), 403
+
+        # Check app from db
+        app = App.get_by_id(app_id)
+
+        if not app:
+            return dict(
+                status='fail',
+                message=f'app {app_id} not found'
+            ), 404
+
+        # Get current time
+        current_time = datetime.datetime.now()
+        yesterday = current_time + datetime.timedelta(days=-1)
+        namespace = project.alias
+        app_alias = app.alias
+
+        prometheus = Prometheus()
+
+        start = validated_query_data.get('start', yesterday.timestamp())
+        end = validated_query_data.get('end', current_time.timestamp())
+        step = validated_query_data.get('step', '1h')
+
+        prom_data = prometheus.query_rang(
+            start=start,
+            end=end,
+            step=step,
+            metric='sum(rate(container_cpu_usage_seconds_total{container!="POD", image!="", namespace="' +
+            namespace+'", pod=~"'+app_alias+'.*"}[5m]))'
+        )
+        #  change array values to json"values"
+        new_data = json.loads(prom_data)
+        cpu_data_list = []
+        try:
+            for value in new_data["data"]["result"][0]["values"]:
+                case = {'timestamp': value[0], 'value': value[1]}
```

Comment:
don't we have to convert these values to floats for ease of use by frontend

Reply:
I have applied the changes